python/ray/tests/test_streaming_generator_2.py

```diff
@@ -531,6 +532,119 @@ def fetch(x):
     assert_no_leak()
 
 
+def test_all_lost_objects_return_errors_if_reconstruction_fails_without_successful_first_attempt(
+    ray_start_cluster,
+):
+    # 1. The streaming generator produces 10 objects in total (called 0, 1, 2, ... , 9) which are stored in plasma
+    # 2. The streaming generator runs on Node A with backpressure=1.
```

Comment:
Is backpressure=1 important for this test? I would suggest making the test fully deterministic if possible by having the actor block forever or exit itself after yielding the first object.

Also a nit, but there's no reason for the actor to yield 10 objects, you could lower it to 2 or 3.